resource-managers/kubernetes/core/src/main/scala/org/apache/spark/deploy/k8s/features/VolcanoFeatureStep.scala

```diff
@@ -41,7 +42,15 @@ private[spark] class VolcanoFeatureStep extends KubernetesDriverCustomFeatureCon
 
   override def getAdditionalPreKubernetesResources(): Seq[HasMetadata] = {
     val client = new DefaultVolcanoClient
-    val template = kubernetesConf.getOption(POD_GROUP_TEMPLATE_FILE_KEY)
+
+    // Currently only driver/job-level PodGroup is supported
+    val template = if (kubernetesConf.isInstanceOf[KubernetesDriverConf]) {
+      kubernetesConf.getOption(POD_GROUP_TEMPLATE_FILE_KEY)
+    } else {
+      throw new SparkException("VolcanoFeatureStep#getAdditionalPreKubernetesResources() is " +
```

Comment:
In this case, `Warning` and `No-Op` will be better.

Reply:
Agreed!